c10/core/TensorImpl.cpp

```diff
@@ -44,6 +44,17 @@ const at::Tensor& TensorImpl::grad() const {
   return autograd_meta_->grad();
 }
 
+const at::Tensor& TensorImpl::fw_grad(uint64_t level, const at::Tensor& self) const {
+  // See TensorImpl::grad() above for explanation about the line below
+  if (!autograd_meta_) return impl::GetAutogradMetaFactory()->undefined_tensor();
```

Comment:
@swolchok You don't happen to know a good way to avoid this goofiness? (I guess the big problem is we're returning `const at::Tensor&` const reference)